aya/src/programs/lirc_mode2.rs

```diff
@@ -101,13 +107,13 @@ impl LircMode2 {
                 io_error,
             })?;
 
-            prog_fds.push(fd as RawFd);
+            prog_fds.push(fd);
         }
 
-        Ok(prog_fds
+        prog_fds
             .into_iter()
-            .map(|prog_fd| LircLink::new(prog_fd, target_fd.as_raw_fd()))
-            .collect())
+            .map(|prog_fd| Ok(LircLink::new(prog_fd, target_fd.try_clone_to_owned()?)))
```

Comment:
prior to this PR we would leak the `prog_fd` as it was created by `bpf_prog_get_fd_by_id` but never closed.